File: shared-atlas-dart/lib/src/interfaces/update_note_dto.dart
class UpdateNoteDto {
  final String? title;
  final String? coverImage;
  final String? icon;

  UpdateNoteDto({
    this.title,
    this.coverImage,
    this.icon,
  });

  factory UpdateNoteDto.fromJson(Map<String, dynamic> json) {
    return UpdateNoteDto(
      title: json['title'] as String?,
      coverImage: json['coverImage'] as String?,
      icon: json['icon'] as String?,
    );
  }
  Map<String, dynamic> toJson() {
    return {
      'title': this.title,
      'coverImage': this.coverImage,
      'icon': this.icon,
    };
  }
}

File: packages/shared-atlas-rust/src/interfaces/update_note_dto.rs
use serde::{Serialize, Deserialize};

/// Payload for updating note metadata.
/// 更新笔记元数据的请求体。
/// 
/// API: `PATCH /api/notes/:id`
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct UpdateNoteDto {
/// Optional new title.
/// 可选的新标题。
    pub title: Option<String>,
/// Optional new cover image URL.
/// 可选的新封面图片URL。
    pub cover_image: Option<String>,
/// Optional new icon.
/// 可选的新图标。
    pub icon: Option<String>,
}